test/eth/Cdp.spec.js

```diff
@@ -11,60 +11,78 @@ import {
 import { mineBlocks } from '../helpers/transactionConfirmation';
 
-let cdpService, cdp, currentAccount, dai, dsProxyAddress, smartContractService;
+let cdpService,
+  cdp,
+  currentAccount,
+  dai,
+  dsProxyAddress,
+  proxyAccount,
+  proxyKey,
+  transactionCount;
 
 // this function should be called again after reverting a snapshot; otherwise,
 // you may get errors about account and transaction nonces not matching.
-async function init() {
+async function init(proxy = false) {
   cdpService = buildTestEthereumCdpService();
   await cdpService.manager().authenticate();
-  currentAccount = cdpService
+  if (proxy) {
+    await setProxyAccount();
+  } else {
+    currentAccount = cdpService
+      .get('token')
+      .get('web3')
+      .currentAccount();
+  }
+}
+
+async function setProxyAccount(newAccount = false) {
+  const accountService = cdpService
     .get('token')
     .get('web3')
-    .currentAccount();
-  if (cdp) cdp._cdpService = cdpService;
+    .get('accounts');
+  const account = newAccount
+    ? {
+        address: newAccount.address,
+        key: newAccount.key
+      }
+    : {
+        address: proxyAccount,
+        key: proxyKey
+      };
+
+  await accountService.addAccount(account.address, {
+    type: 'privateKey',
+    key: account.key
+  });
+  accountService.useAccount(account.address);
+  currentAccount = account.address;
 }
 
-beforeAll(async () => {
-  await init();
-  dai = cdpService.get('token').getToken(DAI);
-  smartContractService = cdpService.get('smartContract');
-
-  // Clear owner of DSProxy created during testchain deployment
-  // (allowing us to create new DSProxy instances using the default address)
-  const owner = await getDsProxyOwner(testnetAddresses.DS_PROXY);
-  if (owner !== '0x0000000000000000000000000000000000000000') {
-    await clearDsProxyOwner(testnetAddresses.DS_PROXY);
-  }
-});
-
-afterAll(async () => {
-  await clearDsProxyOwner(dsProxyAddress);
-});
-
-function getDsProxy(address) {
-  return smartContractService.getContractByAddressAndAbi(
-    address,
-    dappHub.dsProxy,
-    { name: 'DS_PROXY' }
-  );
+async function transferMkr() {
+  const mkr = cdpService.get('token').getToken(MKR);
+  await mkr.transfer(proxyAccount, MKR(1));
 }
 
-async function getDsProxyOwner(address) {
-  return getDsProxy(address).owner();
+function setExistingAccount(name) {
+  const accountService = cdpService
+    .get('token')
+    .get('web3')
+    .get('accounts');
+  accountService.useAccount(name);
 }
 
-async function clearDsProxyOwner(address) {
-  await getDsProxy(address).setOwner(
-    '0x0000000000000000000000000000000000000000'
-  );
-}
+beforeAll(async () => {
+  await init();
+  const account = TestAccountProvider.nextAccount();
+  proxyAccount = account.address;
+  proxyKey = account.key;
+  transferMkr();
+  dai = cdpService.get('token').getToken(DAI);
+});
 
-const sharedTests = openCdp => {
+const sharedTests = (openCdp, proxy = false) => {
```

Comment:
please rename the argument to `useProxy` to make it clear to readers that it's a boolean